x/migration/module/cmd/collect_morse_accounts.go

```diff
@@ -231,26 +232,48 @@ func collectInputApplicationStakes(inputState *migrationtypes.MorseStateExport,
 	for exportApplicationIdx, exportApplication := range inputState.AppState.Application.Applications {
 		appAddr := exportApplication.Address.String()
 
-		// DEV_NOTE: An account SHOULD exist for each actor.
 		if !morseWorkspace.hasAccount(appAddr) {
-			return ErrMorseExportState.Wrapf("account not found corresponding to application with address %q", appAddr)
-		}
-
-		appStakeAmtUpokt, ok := cosmosmath.NewIntFromString(exportApplication.StakedTokens)
-		if !ok {
-			return ErrMorseExportState.Wrapf("failed to parse application stake amount %q", exportApplication.StakedTokens)
+			logger.Logger.Warn().
+				Str("app_address", appAddr).
+				Msg("no account found for application")
+
+			// DEV_NOTE: If no auth account was found for this application, create a new one.
+			newMorseAppAuthAccount := &migrationtypes.MorseAuthAccount{
+				Type: "posmint/Account",
+				Value: &migrationtypes.MorseAccount{
+					Address: exportApplication.Address,
+					Coins:   []cosmostypes.Coin{},
+				},
+			}
+			if _, _, err := morseWorkspace.addAccount(appAddr, newMorseAppAuthAccount); err != nil {
+				return fmt.Errorf(
+					"adding application account to account balance of address %q: %w",
+					appAddr, err,
+				)
+			}
 		}
 
-		if err := morseWorkspace.addAppStake(appAddr, appStakeAmtUpokt); err != nil {
-			return fmt.Errorf(
-				"adding application stake amount to account balance of address %q: %w",
-				appAddr, err,
-			)
+		if exportApplication.StakedTokens != "" {
+			appStakeAmtUpokt, ok := cosmosmath.NewIntFromString(exportApplication.StakedTokens)
+			if !ok {
+				return ErrMorseExportState.Wrapf("failed to parse application stake amount %q", exportApplication.StakedTokens)
+			}
+
+			if err := morseWorkspace.addAppStake(appAddr, appStakeAmtUpokt); err != nil {
+				return fmt.Errorf(
+					"adding application stake amount to account balance of address %q: %w",
+					appAddr, err,
+				)
+			}
+
+			morseWorkspace.accumulatedTotalAppStake = morseWorkspace.accumulatedTotalAppStake.Add(appStakeAmtUpokt)
+			morseWorkspace.numApplications++
+		} else {
+			logger.Logger.Warn().
+				Str("app_address", appAddr).
+				Msg("account stakes as BOTH an application AND a supplier; this account will be unclaimable")
```

Comment:
Can you #PUC error messages OR add a comment explaining why `StakedTokens != ""` is a result of staking both application AND a supplier?

It's not very clear to me.

Ditto below.